### PR TITLE
[Feature] LCX adapter

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description
Adds a new adapter for LCX to the following prices:
- BTC -> USD
- BTC -> EUR
- ETH -> USD
- ETH -> EUR

### Ticket
https://www.pivotaltracker.com/story/show/175505362

### Caveats
@krebernisak there is a a linter warning getting thrown from the example adapter's `CustomError` having the any type on its parameter. I'm sure we will convert the external adapter lib code to TS soon. I can help on this if needed. Just figured I'd mention it.

### CODE REVIEW

File: lcx/src/adapter.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Execute } from '@chainlink/types'
+import { Requester, Validator } from '@chainlink/external-adapter'
+
+const customError = (data: any) => {
+  return data.Response === 'Request failed with status code 400'
```
Comment (krebernisak) [inline lcx/src/adapter.ts:5]: I do not think this is necessary in this case. The Requestor will actually throw on error and this function will not be called.

This custom handler is for cases where API returns 200, but we don't get the expected data in the response so we force an error. I think we should rarely use it, as in most cases most APIs will return the correct code (non 200) for error cases, and our requester will throw an error.

**Review by krebernisak — CHANGES_REQUESTED** [review]
Please add the adapter to the CHANGELOG.

Regarding the TS warnings, they are legitimate and we should slowly start to fix them one by one. We are currently in this transition period where part of the codebase is JS and the other part is TS, so it's acceptable to have some warnings. 

Hopefully, we'll complete this transition soon.

File: lcx/src/adapter.ts (hunk 1 above)
Comment (justinkaseman) [reply lcx/src/adapter.ts]: Good to know, thanks. Removed the `customError`.

File: lcx/src/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export const execute: Execute = async (input) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const base = validator.validated.data.base.toUpperCase()
+  const quote = validator.validated.data.quote.toUpperCase()
+  const url = `https://rp.lcx.com/v1/rates/current/?coin=${base}&currency=${quote}`
```
Comment (boxhock) [inline lcx/src/adapter.ts:17]: We should be able to add these params as an object in the config.

**Review by boxhock — CHANGES_REQUESTED** [review]

**Review by boxhock — DISMISSED** [review]

**Review by krebernisak — APPROVED** [review]

**Review by krebernisak — APPROVED** [review]

**Review by boxhock — APPROVED** [review]

**Pull request merged**